### PR TITLE
Tunnel UDP SRC PORT security

### PR DESCRIPTION
**Pull request by marian-pritsak**
Question to community: Is this feature supported natively on your ASICs, or is it rather something that will have to be done with ACL?

### CODE REVIEW

File: inc/saitunnel.h [hunk 1]
```diff
@@ -735,6 +735,20 @@ typedef enum _sai_tunnel_attr_t
...
     SAI_TUNNEL_ATTR_DECAP_QOS_TC_TO_PRIORITY_GROUP_MAP,
+    
+    /**
+     * @brief Drop tunnel packets with not allowed UDP source port
+     *
+     * Upon enabling this feature, if the tunnel packet ingresses with
+     * UDP source port outside of range defined for this tunnel, it
+     * will be dropped.
+     *
+     * @type sai_bool
```
Comment (kcudnik) [inline inc/saitunnel.h:746]: bool

**Review by kcudnik — CHANGES_REQUESTED** [review]

**Review by kcudnik — APPROVED** [review]

**Comment by rlhui** [comment]
@JaiOCP , @ashutosh-agrawal - please help review.

**Review by ashutosh-agrawal — APPROVED** [review]

**Comment by rlhui** [comment]
@marian-pritsak  - please resolve the DCO error, so that we can merge this, thanks.

**Comment by rlhui** [comment]
@marian-pritsak , please address DCO error. Thanks.

**Comment by rlhui** [comment]
@marian-pritsak  - reminder to resolve DCO error

File: inc/saitunnel.h [hunk 2]
```diff
@@ -736,6 +736,20 @@ typedef enum _sai_tunnel_attr_t
...
+     * Upon enabling this feature, if the tunnel packet ingresses with
+     * UDP source port outside of range defined for this tunnel, it
+     * will be dropped.
+     *
+     * @type bool
+     * @flags CREATE_AND_SET
+     * @default false
+     * @validonly SAI_TUNNEL_ATTR_TYPE == SAI_TUNNEL_TYPE_VXLAN and SAI_TUNNEL_ATTR_VXLAN_UDP_SPORT_MODE == SAI_TUNNEL_VXLAN_UDP_SPORT_MODE_USER_DEFINED
+     */
+    SAI_TUNNEL_ATTR_VXLAN_UDP_SPORT_SECURITY,
```
[the remark this answers is not in the document]
Comment (marian-pritsak) [reply inc/saitunnel.h:751]: Done

**Review by kcudnik — APPROVED** [review]

**Pull request merged**